testing/functional/tests/cypress/integration/institute/03-institution-profile.spec.ts

```diff
@@ -0,0 +1,141 @@
+    cy.visit(url);
+    cy.intercept("GET", "**/bceid-account").as("bceidAccount");
+    loginInstituteObject.loginWithBCEID().should("be.visible").click();
+    loginInstituteObject.loginInWithBCEIDtext().should("be.visible");
+    loginInstituteObject
+      .bceidInputText()
+      .type(username)
+      .should("have.value", username);
+    loginInstituteObject
+      .passwordInputText()
+      .type(password)
+      .should("have.value", password);
+    loginInstituteObject.continueButton().click();
+    cy.wait("@bceidAccount");
+  });
+
+  it("Verify that user redirect to institute profile page", () => {
+    dashboardInstituteObject.dashboardButton().click();
+    dashboardInstituteObject.manageInstitutionButton().click();
+    dashboardInstituteObject.locationVerifyText().should("be.visible");
+    institutionObject.institutionDetailsButton().click();
+    institutionObject.institutionInformationText().should("be.visible");
+  });
+
+  it("Check that legal operating name text field should be disable or not", () => {
+    dashboardInstituteObject.dashboardButton().click();
+    dashboardInstituteObject.manageInstitutionButton().click();
+    dashboardInstituteObject.locationVerifyText().should("be.visible");
+    institutionObject.institutionDetailsButton().click();
+    institutionObject.legalOperatingNameInputText().should("be.disabled");
+  });
+
+  it("Clicking on the save button without filling out the required fields", () => {
+    dashboardInstituteObject.dashboardButton().click();
+    dashboardInstituteObject.manageInstitutionButton().click();
+    dashboardInstituteObject.locationVerifyText().should("be.visible");
+    institutionObject.institutionDetailsButton().click();
+    institutionObject.primaryEmailInputText().clear();
+    cy.wait(2000);
+    institutionObject.submitButton().click();
+  });
+
+  it(
+    "Check that when user enter data only in non mandatory field and click on next section.",
+    { retries: 4 },
+    () => {
+      dashboardInstituteObject.dashboardButton().click();
+      dashboardInstituteObject.manageInstitutionButton().click();
+      dashboardInstituteObject.locationVerifyText().should("be.visible");
+      institutionObject.institutionDetailsButton().click();
+      cy.reload();
+      institutionObject.primaryEmailInputText().clear();
+      institutionObject.primaryEmailInputText().clear();
+      cy.wait(2000);
+      institutionObject.submitButton().click();
+      institutionObject.unexpectedErrorMessage().should("be.visible");
+    }
+  );
+
+  it("Check when user enter data only in mandatory field then able to save the details or not.", () => {
+    cy.fixture("institutionProfileData").then((data) => {
+      dashboardInstituteObject.dashboardButton().click();
+      dashboardInstituteObject.manageInstitutionButton().click();
+      dashboardInstituteObject.locationVerifyText().should("be.visible");
+      institutionObject.institutionDetailsButton().click();
+      cy.wait(1000);
```

Comment:
Is there a reason to use the `cy.wait(1000)` with a specific timeout here?

Reply:
https://github.com/bcgov/SIMS/pull/1151#discussion_r848486536

Reply:
Added wait for a specific route instead of using specific amount of time.